### ISSUE TITLE
Runs out of memory if num_tasks not mentioned.

### ISSUE THREAD

**Issue by debajyotidatta** [learnables/learn2learn#70]
Since this generates all possible tasks and making this into a list and not through an iterator means it will run out of memory.

---------------------------------------------------------------------------
```
MemoryError                               Traceback (most recent call last)
<ipython-input-2-497a9350dcca> in <module>
    292     loss_fn = torch.nn.NLLLoss().cuda()
    293 
--> 294     eval_generator = l2l.data.TaskGenerator(dataset=omniglot, ways=args.k_test)
    295     support_t = eval_generator.sample(shots=args.q_test)
    296     query_t = eval_generator.sample(shots=args.q_test)

/data/home/debo/l2l/learn2learn/learn2learn/data/task_generator.py in __init__(self, dataset, ways, shots, classes, tasks)
    147 
    148         if tasks is None:
--> 149             self.tasks = list(permutations(self.classes, self.ways))
    150         elif isinstance(tasks, int):
    151             self.tasks = self.generate_n_tasks(tasks)

MemoryError: 
```

**Comment by praateekmahajan** [comment]
Will be resolved by #78

**Comment by praateekmahajan** [comment]
PR merged :78
